components/director/run.sh

```diff
@@ -121,12 +131,83 @@ else
 
     echo -e "${GREEN}Populate DB${NC}"
 
-    cat ${ROOT_PATH}/../schema-migrator/migrations/director/*.up.sql | \
-        docker exec -i ${POSTGRES_CONTAINER} psql -U "${DB_USER}" -h "${DB_HOST}" -p "${DB_PORT}" -d "${DB_NAME}"
-    cat ${ROOT_PATH}/../schema-migrator/seeds/director/*.sql | \
-        docker exec -i ${POSTGRES_CONTAINER} psql -U "${DB_USER}" -h "${DB_HOST}" -p "${DB_PORT}" -d "${DB_NAME}"
+    if [[ ${DUMP_DB} = false ]]; then
+        cat ${ROOT_PATH}/../schema-migrator/migrations/director/*.up.sql | \
+            docker exec -i ${POSTGRES_CONTAINER} psql -U "${DB_USER}" -h "${DB_HOST}" -p "${DB_PORT}" -d "${DB_NAME}"
+
+        cat ${ROOT_PATH}/../schema-migrator/seeds/director/*.sql | \
+            docker exec -i ${POSTGRES_CONTAINER} psql -U "${DB_USER}" -h "${DB_HOST}" -p "${DB_PORT}" -d "${DB_NAME}"
+    else
+        if [[ -f ${ROOT_PATH}/../schema-migrator/seeds/dump.sql ]]; then
+            echo -e "${GREEN}Will reuse existing dump in schema-migrator/seeds/dump.sql ${NC}"
+        else
+            echo -e "${YELLOW}Warning: Please ensure that your kubectl context points to an existing Compass development cluster${NC}"
+            CURRENT_KUBE_CONTEXT=$(kubectl config current-context)
+
+            if [[ $CURRENT_KUBE_CONTEXT != *dev* ]]; then
+                echo -e "${RED}Error: Current kubectl context does not point to an existing Compass development cluster${NC}"
+                exit 1
+            fi
+
+            sleep 2
+            echo -e "${GREEN}Will dump and use database data from connected kubernetes Compass development cluster${NC}"
+
+            REMOTE_DB_PWD=$(base64 -d <<< $(kubectl get secret -n compass-system compass-postgresql -o=jsonpath="{.data['postgresql-director-password']}"))
+            DIRECTOR_POD_NAME=$(kubectl get pods -n compass-system | grep "director" | head -1 | cut -c -33)
+            kubectl port-forward --namespace compass-system $DIRECTOR_POD_NAME 5555:5432 &
+            sleep 5 # necessary for the port-forward to open in time for the next command
+
+            echo -e "${GREEN}Dumping database...${NC}"
+
+            trap cleanup_temp_migrations EXIT
+
+            PGPASSWORD=$REMOTE_DB_PWD pg_dump --dbname=director --file=${ROOT_PATH}/../schema-migrator/seeds/dump.sql --host=localhost --port=5555 --username=director --column-inserts --no-owner --no-privileges
+
+            echo -e "${GREEN}Database dumped!${NC}"
+
+            pkill kubectl
+        fi
+
+        cat ${ROOT_PATH}/../schema-migrator/seeds/dump.sql | \
```

Comment:
if the db on dev is on different version than the local db, it would be good to atleast get some warning that  the data  import might fail to save someone troubleshooting time. Alternatively you might fix it so that if the 2 db versions differ, you might "roll back"(apply only up to) the local db to match the remote db before the import, do the data import and than roll up the extra migrations ontop - this might even be used to kind of quickly verify that the new migrations can be properly applied with existing data while starting up the local dev (a useful side effect?). Then the minikube PR prow job (if we configure it to do dump importing) will actually always verify that the new migrations are OK. 
Anyway, I think we don't need to spin up gke clusters for this so prolly using minikube should be fine/faster esp as its just migration stuff.

Reply:
It will not be uncommon for local db to mismatch dev db during development. If a mismatch exists it will be pretty much always in favor of the local db which has newer migrations (based on latest master + current feature that's being developed).
Having said that I can't think of a scenario where the newer migrations when applied on a db based on the migrations to that point + corresponding data (talking about the db dump from dev here) might fail unless the migrations themselves are corrupt, which should error anyway?

Reply:
if a migration introduces a constraint which is not compatible with the data, the migration will fail. The point was that one option to detect this would be in run.sh

Reply:
I guess I went with too few words - there were two parts of the proposal (not sure if it makes sense, you tell me :) )
1. If you do the adjustments in `this` run.sh when one boots up locally his director with dump-data parameter he will get out of the box validation for his new migrations.
2.  if you do the same in the `installation/cmd` run.sh the minikube prow job will also do this same validation on PR. (Here some ip whitelisting in gcp might be needed)

Reply:
The validation behaviour that you described above is the exact same way that it is currently implemented. The DB dump file contains not only `INSERT` statements which import the data, but it also includes DDL statements which create all tables, constraints etc.
So applying the dump file to a local DB would essentially mean that we apply all migrations up to the point of the dump along with it's data, and once that's done, we leverage the migrate tool to apply all new migrations and thus validating that any introduced constraints properly work with the existing data we have on dev.

Reply:
I will however add a sort of warning to notify when the two versions mismatch. While testing this, I had the unfortunate time wasting situation where the local migrations (which presumably should be ahead of dev migrations) was actually behind dev's migrations - this happens when you work on a branch for some period without merging the latest master changes..

The `migrate` tool was returning an error with a somewhat unintuitive message and it didn't make sense until I realized what was happening. This was the error message: `error: file does not exist`